src/NHibernate/Bytecode/Lightweight/ReflectionOptimizer.cs

```diff
@@ -101,13 +101,22 @@ protected virtual void ThrowExceptionForNoDefaultCtor(System.Type type)
 
 		protected DynamicMethod CreateDynamicMethod(System.Type returnType, System.Type[] argumentTypes)
 		{
-			System.Type owner = mappedType.IsInterface ? typeof (object) : mappedType;
-#pragma warning disable 612,618
-			bool canSkipChecks = SecurityManager.IsGranted(new ReflectionPermission(ReflectionPermissionFlag.MemberAccess));
-#pragma warning restore 612,618
+			var owner = mappedType.IsInterface ? typeof (object) : mappedType;
+			var canSkipChecks = CanSkipVisibilityChecks();
 			return new DynamicMethod(string.Empty, returnType, argumentTypes, owner, canSkipChecks);
 		}
 
+		private static bool CanSkipVisibilityChecks()
+		{
+#if NETFX
+			var permissionSet = new PermissionSet(PermissionState.None);
+			permissionSet.AddPermission(new ReflectionPermission(ReflectionPermissionFlag.MemberAccess));
+			return permissionSet.IsSubsetOf(AppDomain.CurrentDomain.PermissionSet);
+#else
+			return false;
```

Comment:
As far as I understand it, under .Net Core, we should always have the permission. But with .Net Standard, hard to know.